WcaOnRails/app/models/registration.rb

```diff
@@ -109,15 +109,22 @@ def entry_fee
   def paid_entry_fees
     Money.new(
       registration_payments.sum(:amount_lowest_denomination),
-      # FIXME: see https://github.com/thewca/worldcubeassociation.org/issues/1257
-      (competition.currency_code.blank? ? Money.default_currency : competition.currency_code),
+      competition.currency_code,
     )
   end
 
   def outstanding_entry_fees
     entry_fee.nil? ? 0 : entry_fee - paid_entry_fees
   end
 
+  def has_to_pay_fee_here?
+    (pending? || accepted?) && competition.connected_stripe_account_id && outstanding_entry_fees > 0
```

Comment:
`.connected_stripe_account_id` is how we check if a competition is using WCA registration? I wonder if it would make sense to add a helper method to `Competition` with a clearer name, although I'm blanking on what such a name would be... maybe `.using_online_payments?`